modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -253,6 +253,9 @@ bool TiffDecoder::readHeader()
             int wanted_channels = normalizeChannelsNumber(ncn);
             switch(bpp)
             {
+                case 1:
+                    result = true;
+                    break;
```

Comment:
What is about "m_type" ?
It must be initialized in this method.

Also it would be nice to add proper error reporting via "default:" case.

Reply:
It was not previously initialized so tbh I have not figured it out.

Yes I agree the default case should have some error message because now it silently failed and it was frustrating.

I will figure these things out when I can.

Reply:
Actually this case was not properly supported (bpp=1 case works "by mistake").
Also there are no tests for this type of images (btw, imgcodecs test data is [here](https://github.com/opencv/opencv_extra/tree/3.4/testdata/highgui/readwrite)).

Reply:
Yup I know it worked by mistake, just wanted to revert the changes that broke this functionality which the original committer was unwilling to do :)

So this will take some time on my behalf...